File: API/Helpers/UserParams.cs
namespace API.Helpers
{
    public class UserParams : PaginationParams
    {
        public string? Gender { get; set; }

        public int? CurrentUserId { get; set; }

        public int MinExperience { get; set; } = 1;

        public int MaxExperience { get; set; } = 15;

        public string OrderBy { get; set; } = "lastActive";

    }
}

File: client/src/app/_models/userParams.ts
import { User } from "./user";

export class UserParams {
  gender: string;
  minExperience = 1;
  maxExperience = 15;
  pageNumber = 1;
  pageSize = 5;
  orderBy = 'lastActive';

  // makes sure the opposite gender is set , from our current user
  // not utilized in refactor
  constructor(user: User | null){
    this.gender = user?.gender === 'female' ? 'male' : 'female'
  }
}